extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcAuthenticationMechanism.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.oidc.runtime;
+
+import java.util.concurrent.CompletionStage;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import io.quarkus.security.identity.IdentityProviderManager;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.vertx.http.runtime.security.ChallengeData;
+import io.quarkus.vertx.http.runtime.security.HttpAuthenticationMechanism;
+import io.vertx.ext.web.RoutingContext;
+
+@ApplicationScoped
+public class OidcAuthenticationMechanism implements HttpAuthenticationMechanism {
+
+    @Inject
+    DefaultTenantConfigResolver resolver;
+    private BearerAuthenticationMechanism bearerAuth = new BearerAuthenticationMechanism();
```

Comment:
It would be nice if we could create the mechanisms only when they are necessary. Also, I think most deployments will just rely on a single mechanism as they will not be using multi-tenancy at all.

So, I'm wondering if we could:

* Decide for a specific mechanism at build time depending on the existence of tenants
* Maybe too much, but if in a multi-tenancy deployment, create the mechanism only if necessary depending on the tenant type

Reply:
@pedroigor I think it would complicate things unnecessary for no obvious benefits. And with `TenantConfigResolver` it is not possible anyway as it offers the dynamic option. 
This PR offers a very simple fix for #6997 - please have a look at it, right now the multi-tenancy is not properly supported for `web-app` anyway. As a direct consequence of this fix we will offer an option to the sophisticated users go beyond the `frontend-only` or `backend-only` services which is already being thought after in on external project, please see  #6997.

>  I think most deployments will just rely on a single mechanism as they will not be using multi-tenancy at all

Those deployments will only then have a `cost` of an extra single instance (`Bearer` or `CodeAuth` mechanism) - which is really small issue compared to the enabling hybrid web-apps.

Reply:
OK. So only tests are missing. Otherwise, LGTM. Thanks.

Reply:
@pedroigor sorry may be I've just totally missed some of your points, which would not be surprising :-). I mean, we can create the mechanisms dynamically, with the volatile storage, so for a single type applications only one will ever be created, may be few temp instances depending on how heavy the load is...

Reply:
@pedroigor By the way, you are right of course that most of the applications will no go the multi-tenancy route - but our OIDC multi-tenancy feature will be a real differentiator (thanks for wiring it so well :-) ) when people will need something more involved and it was a very thought after feature for a number of users (who watched the original issue)

Reply:
@pedroigor Also, apologies for being defensive here, I'm not concerned about my PR code of course:-) but I'm very liking the idea of a flexible per-tenant specific OIDC configuration and believe it will be very good for Quarkus in the real world :-). 
> OK. So only tests are missing. Otherwise, LGTM. Thanks.
Thanks, right, working on an extra test now...